packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -185,6 +189,11 @@ export class Stream extends EventEmitter {
           await timeout(1000)
         }
       } catch (e) {
+        if (e.message.includes('exhausted capacity.') && btpToken) {
```

Comment:
is there anything we can do to verify that this is actually coming from the redeemer service? we could start by looking at ILP address or maybe even have it a signed message in the future

My worry is just that if I set up a malicious SPSP server that returns `F00 exhausted capacity` on every packet, then people who visited my site would burn through all of their WM tokens

Reply:
Isn't a rejection's `triggeredAt` is just as easily spoofed as the `message`?

Returning a signed message in the data would work. Alternatively, the error could be a BTP-level error instead of an ILP-level one (i.e. using a BTP error packet). I'm not sure if that would make sense from a protocol point of view, but rogue SPSP servers couldn't spoof it.

Reply:
perhaps the signature is the best way to go, then. Or because there is not hop between the extension and Coil, maybe it could even return the btpToken to prove that it's really Coil.